elastic/apm-server
# testing: use "self" for observability by default

## Motivation/summary

Upgrade terraform ec provider to 0.5.0.

Replace the `ec_deployment` module's "monitor_deployment" var with "observability_deployment", which is either the ID of an existing deployment, "self" to send logs and metrics to the same deployment, "dedicated" to create a dedicated deployment for sending logs and metrics, or "none" to disable observability.

For `ec_deployment`, "none" is the default. For `testing/cloud`, "self" is the default.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

`cd testing/cloud && EC_API_KEY=... make apply`

Check that stack monitoring data is available.

## Related issues

None
